### ISSUE TITLE
loss -nan

### ISSUE THREAD

**Issue by bubulv** [pjreddie/darknet#567]
I use yolov3 in my dataset.  I found in yolo_layer.c  function forward_yolo_layer(), the count=0 and class_count=0 lead to the loss = '-nan'. 
  
" int mask_n = int_index(l.mask, best_n, l.n);"

the mask_n often equal -1, so the count=0.

in yolov3-voc.fcg:

[yolo]
mask = 3,4,5
anchors = 10,13,  16,30,  33,23,  30,61,  62,45,  59,119,  116,90,  156,198,  373,326
classes=2
num=9
jitter=.3
ignore_thresh = .5
truth_thresh = 1
random=1

I don't know how the mask effect?  How did I set on my own dataset?

**Comment by txdat** [comment]
according to paper, each yolo (detection) layer get 3 anchors with associated with its size, mask is selected anchor indices.

**Comment by yqyao** [comment]
I meet the same question too, I just use the voc datasets.

**Comment by ciwei123** [comment]
I meet the same question too,when I just use the voc datasets.

**Comment by bubulv** [comment]
@ciwei123 @yqyao  -nan is no effect. I trained 900.weights  to test some images, the prediction is ok. you can have a try. I didn't change any thing.

**Comment by raulincze** [comment]
Hello,

I've encountered the same problem when training on a custom dataset. This might happen because you changed the batch and subdivision numbers for training. 

While I don't have a theoretical understanding yet, here's what empirically worked for me: tweak the learning rate and the burn_in numbers. 

With a batch of 2, I've set the learning rate to 0.0001 and the burn_in to 10000 and the issue seems to be gone with the loss steadily converging.
